### PR TITLE
Adds builder as separate package

### PR DESCRIPTION
**Pull request by NothingEverHappens**

### CODE REVIEW

**Comment by kirjs** [comment]
Now builder is treated as a separate package an has it's own binary.

This is how you debug it:
`node --inspect-brk ./node_modules/.bin/builder-30-seconds serve`

**Comment by fetis** [comment]
> node --inspect-brk ./node_modules/.bin/builder-30-seconds serve

I think we need to add `DEVELOPMENT.md` and add this line there

File: builder/.bin/bin.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env node
+const path = require('path');
+
+const CONFIG_FILE_NAME = '30s.config.json';
+
+require('yargs')
```
Comment (fetis) [inline builder/.bin/bin.js]: I suggest to bring some order here, move `require` on top next to other requires and move `getConfig` before commands parsing to make structure more linear

File: builder/.bin/bin.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  .command('serve', 'start a server', () => {
+    const serve = require('../serve');
+    const config = getConfig();
+    config.metadata = {
```
Comment (fetis) [inline builder/.bin/bin.js:21]: this can be move to `getConfig` method

File: builder/config.js [hunk 3]
```diff
@@ -6,6 +6,7 @@ const collections = require('metalsmith-collections');
 const discoverPartials = require('metalsmith-discover-partials');
 const registerHelpers = require('metalsmith-register-helpers');
 const metalsmithStatic = require('metalsmith-static');
+const metalsmithLocalStatic = require('./plugins/local-static');
```
Comment (fetis) [inline builder/config.js]: plz move it below to "our stuff"

File: builder/package.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "builder-30-seconds",
+  "version": "1.0.0",
+  "description": "",
```
Comment (fetis) [inline builder/package.json]:
```suggestion
  "description": "30 seconds site generator",
```

File: config/30s.config.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "30 Seconds of Angular",
+  "snippets": "./snippets",
+  "static": "./config/static",
+  "metadata": {
+    "base": "https://30.codelab.fun/",
+    "codeRunnerHost": "https://codelab.fun",
+    "title": "30 Seconds of Angular",
+    "twitter": "30sec_angular",
+    "isDevMode": false,
```
Comment (fetis) [inline config/30s.config.json]: I think we can skip explicit `false` value

File: typesrcipt/config/30s.config.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "30 Seconds of TypeScript",
+  "snippets": "./snippets",
+  "static": "./config/static",
+  "metadata": {
+    "base": "https://30.codelab.fun/",
+    "codeRunnerHost": "https://codelab.fun",
+    "title": "30 Seconds of TypeScript",
+    "twitter": "30sec_angular",
+    "isDevMode": false,
```
Comment (fetis) [inline typesrcipt/config/30s.config.json]: same her about `false`

File: builder/plugins/local-static.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
+/**
+ * Module Dependencies.
+ */
+
+var debug = require('debug')('metalsmith-static');
+var path  = require('path');
```
Comment (fetis) [inline builder/plugins/local-static.js]:
```suggestion
var path = require('path');
```

File: builder/plugins/local-static.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+var fsextra    = require('fs-extra');
```
Comment (fetis) [inline builder/plugins/local-static.js]:
```suggestion
var fsextra = require('fs-extra');
```

File: builder/plugins/local-static.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+  return function(files, metalsmith, done) {
+    assets.forEach(function(opts) {
+      var src = opts.src || 'public';
+      var relDest = opts.dest || 'public';
+      var createDest = opts.createDest || true;
+
+      var dst = path.join(metalsmith.destination(), relDest);
+
+      debugger
```
Comment (fetis) [inline builder/plugins/local-static.js]: plz remove `debugger` statements

File: builder/plugins/local-static.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+      if (createDest) {
+        var dir = path.dirname(dst);
+
+        debug('creating: %s', dir);
+        fsextra.mkdirpSync(dir);
+        fsextra.mkdirpSync(dst);
+      }
+      debugger
+
+
```
Comment (fetis) [inline builder/plugins/local-static.js:32]:
```suggestion

```

**Review by fetis — APPROVED** [review]
generally, looks good to me.

I think 2 things missed (can be added in next PRs)
1. styles are still served from builder directory
1. instead of picking up config from `config` folder, specify file as parameter with some defaults

**Pull request merged**